docs/it-manual/api/program-applications.md

```diff
@@ -1,182 +1,781 @@
+The question key is generated from the immutable admin name of the question. The immutable admin name of the question is specified by the admin when creating a question, and the question key is visible in the [generated API docs](#generated-program-specific-docs) page.
+
+The question key is generated by removing non-letter characters and replacing spaces with underscores, so `applicant name4?` becomes `applicant_name`.
+
+For [enumerator](/docs/user-manual/civiform-admin-guide/using-enumerator-questions-and-screens-in-a-program.md) questions, such as "List all the members of your household?" (which might have a question key of `household_members`), and "What are each of their favorite colors?" (question key of `household_member_favorite_color`), the repeated question is nested under the enumerator question like this:
+
+{% code %}
+```json
+{
+  "household_members": {
+    "question_type": "ENUMERATOR",
+    "entities": [
+      {
+        "entity_name": "Taylor",
+        "household_member_favorite_color": {
+          "question_type": "TEXT",
+          "text": "purple"
+        }
+      },
+      {
+        "entity_name": "Sza",
+        "household_member_favorite_color": {
+          "question_type": "TEXT",
+          "text": "purple"
+        }
+      },
+      {
+        "entity_name": "Carley Rae",
+        "household_member_favorite_color": {
+          "question_type": "TEXT",
+          "text": "red"
+        }
+      },
+    ]
+  },
+}
+```
+{% endcode %}
+
+In this example, to access the favorite color of the third household member, you'd use this path through the `application` object:
+
+{% code %}
+```js
+application.household_members.entities[2].household_member_favorite_color.text
+```
+{% endcode %}
+
+{% hint style="info" %}
+**Note** Questions can be nested arbitrarily deep, so if a question asked for the hours worked at each job a household member has, a path might look like
+{% code %}
+```js
+application.household_members.entities[2].household_member_jobs.entities[1].household_member_jobs_hours_worked.number
+```
+{% endcode %}
+This is the path to the number of hours worked at the 2nd job of the 3rd household member.
+{% endhint %}
+
+See [Enumerator questions](#enumerator-questions) below for more about enumerator questions.
+
+### Question objects
+Question objects contain the applicant's answers to a question, as well as some additional metadata. They all contain the same metadata property, as well as additional properties that vary based on the question type.
+
+{% hint style="warning" %}
+**Warning** The order of properties within the a JSON object is not guaranteed.
+{% endhint %}
+
+#### Question metadata
+All question objects have a `question_type` field.
+
+- **Property**: `question_type`
+- **JSON Type**: `string`
+- **Format**: An enum currently consisting of one of [`ADDRESS`, `MULTI_SELECT`, `CURRENCY`, `DATE`, `SINGLE_SELECT`, `EMAIL`, `ENUMERATOR`, `FILE_UPLOAD`, `ID`, `NAME`, `NUMBER`, `TEXT`, `PHONE`].\
```

Comment:
question (non-blocking): Is there any where a list of these is generated programmatically? Or a list of question types since these correspond to those? It could be helpful to link to that if it exists.

Reply:
added a link!